core/modules/editor/engines/simple.js

```diff
@@ -67,9 +67,7 @@ Set the text of the engine if it doesn't currently have focus
 */
 SimpleEngine.prototype.setText = function(text,type) {
 	if(!this.domNode.isTiddlyWikiFakeDom) {
-		if(this.domNode.ownerDocument.activeElement !== this.domNode || text === "") {
-			this.domNode.value = text;
-		}
+		this.domNode.value = text;
```

Comment:
The test you've removed is intended to prevent text inputs that contain the focus from being updated in response to changes to the underlying tiddler. As I recall, without this check typing into the text input will trigger an update of the text input, losing the cursor position/selection.

Reply:
Thanks @Jermolene. This test somehow blocks updating the input's value by navigating up/down through surrounding keyboard widgets. Updating an input's value with a button widget works fine though. I don't know if it's a bug somewhere or if I need to add another test at this point in the code, I just didn't notice any difference with other inputs with or without this test

Reply:
@Jermolene, the following code doesn't update the input fields' value, that's the problem here:

```
<$keyboard key="alt-shift-A" actions="""<$action-setfield $tiddler="$:/temp/test" text="12345"/>""">
<$edit-text tiddler="$:/temp/test" default="" tag="input" placeholder="">
</$keyboard>
```

Reply:
@Jermolene, I don't notice any downsides by removing this test

Reply:
In my tests, things work OK without that test on Firefox and Chrome, but it doesn't work on Webkit/Safari (Mac and iOS).

The issue is that on Webkit assigning a new value to the input element also moves the cursor to the end of the input text. I appreciate you may not have access to Safari to text it but the steps are:

1. First, make the change in the lines above (ie replace lines 70-72 with line 71)
2. Build the wiki and open it in Safari
3. Click the "new tiddler" button
4. Place the cursor after the word "new" and add the letters "ish"
5. Wait for up to 1 second for the draft debouncing to occur
6. Watch the cursor move to the end of the input box

Back when this code was new, the conditional was needed on Chrome and Firefox too. I think that since then both browsers have added a check to ignore attempts to set a text input to the text that it already contains. This can be verified by changing the line above to something like `this.domNode.value = text + "$";`; in Chrome and Firefox, attempts to type in the middle of a text box will again see the cursor move to the end.

Reply:
Thanks for clarifying this, @Jermolene. Yes, I don't have access to Safari but I may be able to test with a webkit browser on linux. But now I understand the importance of this test.

To make this functionality happen nevertheless I could make the whole macro simpler and renounce to update the inputs' value with the selected dropdown item

Reply:
Changing the test to checking whether the new text is different than the current text seems to work OK on Webkit, and is more readable:

```
if(this.domNode.value !== text || text === "") {
	this.domNode.value = text;
}
```

Reply:
> Changing the test to checking whether the new text is different than the current text seems to work OK on Webkit, and is more readable:
> 
> ```
> if(this.domNode.value !== text || text === "") {
> 	this.domNode.value = text;
> }
> ```

This would make it work :smile:

Reply:
I've updated the demo with your suggestion now, @Jermolene. works fine on firefox and chrome